pyk/src/pyk/proof/implies.py

```diff
@@ -26,13 +26,13 @@
 _LOGGER: Final = logging.getLogger(__name__)
 
 
-@dataclass
-class ImpliesProofStep(ProofStep):
+@dataclass(frozen=True, eq=True)
```

Comment:
```suggestion
@dataclass(frozen=True)
```

(`eq=True` is the default.)

Reply:
Fixed.